distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -78,6 +78,13 @@ public void execute(Terminal terminal, OptionSet options, Environment env, Proce
             return;
         }
 
+        if (Runtime.version().feature() >= 24) {
```

Comment:
Should we have put this in `JavaVersionChecker` instead so it applies to _all_ CLI tools or does that not matter since it's only server that actually attempts to use the security manager?

Reply:
For example https://github.com/elastic/elasticsearch/pull/122233 wasn't backported to 8.17, so we'd fail if folks try to run `elasticsearch-plugin install` on Java 24.